docs/DEVELOPING.md

```diff
@@ -1,20 +1,72 @@
 # Developer Guide
 
+## Architecture Overview
+
+### Code organization
+
+```mermaid
+graph
+  fil-stat-repo(github.com/filecoin-project/filecoin-station) --> fil-stat-main(Station main process)
+  fil-stat-repo --> fil-stat-webui(Station WebUI)
+
+  saturn-l2-repo(https://github.com/filecoin-project/saturn-l2)
+  --> saturn-l2-go(Saturn L2 Node binary)
+
+  saturn-l2-webui-repo(github.com/filecoin-project/saturn-l2-webui)
+  --> saturn-l2-webui(Saturn L2 WebUI)
+  saturn-l2-webui --> saturn-l2-go
+
+  fil-stat-main --> fil-stat-distro(Desktop Application)
+  fil-stat-webui --> fil-stat-distro
+  saturn-l2-go --> fil-stat-distro
+```
+
+Repositories:
+
+- https://github.com/filecoin-project/filecoin-station
+- https://github.com/filecoin-project/saturn-l2
+- https://github.com/filecoin-project/saturn-l2-webui
+
+### End-to-end release workflow
+
+1. `saturn-l2-webui` publishes a new release with WebUI assets
+2. `saturn-l2` is updated to use the new `saturn-l2-webui` version
+3. `saturn-l2` publishes a new release
+4. `filecoin-station` is updated to use the new `saturn-l2 `version
+5. `filecoin-station` publishes a new release
+
 ## Development workflow
 
-Run `npm start` to open the Electron app and load the WebUI via the Vite DEV server. Changes made
-inside `renderer` files will be immediately applied in the running app.
+### Initial setup
+
+1. Install the latest LTS version of Node.js
+2. Clone this repository
+3. Run `npm install` to install all dependencies
+4. Run `npm run build:saturn` to download Saturn L2 Node.
```

Comment:
is this a build step or more just a download step?

Reply:
See https://github.com/filecoin-project/filecoin-station/pull/39